### PR TITLE
poco: propagate dependencies imported by CMake scripts

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

poco's CMake scripts search for a few libraries at build time, so they should be propagated to allow dependent packages to build without manually adding these libraries as `buildInputs`. I enabled split outputs to prevent the runtime closure size from increasing.

The only packages affected by this change are `clickhouse`, which builds successfully, and `toggldesktop`, which was already broken.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lopsided98** [comment]
Actually this is broken. I assumed it worked because clickhouse worked, but I tried another out of tree package and it failed because for some unfathomable reason poco doesn't use GNUInstallDirs.

**Comment by lopsided98** [comment]
I have submitted https://github.com/pocoproject/poco/pull/3105 upstream to make poco use GNUInstallDirs.

I added the patch to this PR.

**Comment by lopsided98** [comment]
cc @orivej

**Comment by mikepurvis** [comment]
I've been using this from @lopsided98's fork for a few weeks and it's worked great for me— I'd love to get it merged.

**Comment by lopsided98** [comment]
I fixed the conflicts.

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64

**Comment by lopsided98** [comment]
> Why omit some of the `CMakeLists.txt`s?

Those files are not included in the release tarball.

**Comment by risicle** [comment]
Cool - that would be nice to include in a comment.

Oh and while I'm making a wishlist, adding a `name = "use-gnuinstalldirs.patch";` to the fetchpatch call and a comment linking to the PR :grin:

File: pkgs/development/libraries/poco/default.nix [hunk 1]
```diff
@@ -10,9 +10,31 @@ stdenv.mkDerivation rec {
     sha256 = "1jilzh0h6ik5lr167nax7q6nrpzxl99p11pkl202ig06pgh32nbz";
   };
 
+  patches = [
+    # Use GNUInstallDirs (https://github.com/pocoproject/poco/pull/3105)
+    (fetchpatch {
+      name = "ure-gnuinstalldirs.patch";
```
Comment (risicle) [inline pkgs/development/libraries/poco/default.nix:16]: `ure` ;)

**Pull request merged**